cmd/client/client.go

```diff
@@ -162,6 +162,9 @@ func Run(ctx context.Context, conf Config) error {
 		if err := client.Start(ctx); err != nil {
 			errCh <- err
 		}
+		if err := client.Error(); err != nil {
```

Comment:
`client.Start` is always going to return an error before you get to this line. It is likely to be a `context.Cancelled` error if one of the streamHandler goroutines hits a different error first. This defeats the purpose of reading `client.Error()` here. What you could do instead is have `client.Start` return `c.Error()`. Any place that Start hits an error replace:

```return err```

with
```
c.cancel(err)
return c.Error()
```

That way `client.Start` always returns the first error it encountered and you will be able to remove the `client.Error()` check here.

Reply:
Actually, a smaller and perhaps better change you could make here is to leave `client.Start` how you currently have it and replace the code here with:

```
c.cancel(client.Start(ctx))
if err := c.Error(); err != nil {
    errCh <- err
}
```
This ensures you always call `cancel` when `Start` returns, which you need to do to clean up any `streamHandler` goroutines and gets you the most recent (most interesting) error.

Reply:
Thanks for the comments. I was thinking of calling `cancel` as a defer within Start, but that will involves slightly more change (similar to your first proposal). I ended up following your most recent proposal, which looks good too.

Reply:
Sorry, to go back-and-forth here. I actually do like your idea better of having cancel called inside `Start`, and having `Start` return `c.Error()`. That way other users of `Start` don't have to reimplement what you do inside `Run`.